feature-utils/poly-look/src/react-components/legend.js

```diff
@@ -0,0 +1,51 @@
+class IncorrectLegendEntry extends Error {
+  constructor(entry) {
+    super(`«${JSON.stringify(entry)}» is not a LegendEntry`);
+  }
+}
+
+class LegendEntry {
+  constructor(description, color) {
+    this._description = description;
+    if (color.match(/^#[0-9a-fA-F]{6}$/)) {
+      this._color = color;
+    } else {
+      throw new IncorrectLegendEntry({
+        description: description,
+        color: color,
+      });
+    }
```

Comment:
```suggestion
    if (!color.match(/^#[0-9a-fA-F]{6}$/)) {
      throw new IncorrectLegendEntry({
        description: description,
        color: color,
      });
    }
    this._color = color;
```
so assertion is first and then action